solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseCloudSolrClient.java

```diff
@@ -838,6 +840,17 @@ public RouteException(ErrorCode errorCode, NamedList<Throwable> throwables, Map<
 
   @Override
   public NamedList<Object> request(@SuppressWarnings({"rawtypes"})SolrRequest request, String collection) throws SolrServerException, IOException {
+    return makeRequest(request, collection, null);
+  }
+
+  /**
+   * Will execute a request synchronously if asyncListener is null, asynchronously otherwise.
+   *
+   * @return A {@link NamedList} with the response if sync, or a {@link NamedList} containing a single Cancellable object if async
+   */
+  NamedList<Object> makeRequest(SolrRequest<?> request,
+                                String collection,
+                                AsyncListener<LBSolrClient.Rsp> asyncListener) throws SolrServerException, IOException {
```

Comment:
Both `LbSolrClient.Rsp` and wrapping `Cancellable` in a `NamedList` seems like a workaround to me. Kinda prefer another option if that exist.